src/components/Site.tsx

```diff
@@ -1,80 +1,30 @@
 import * as React from "react";
-import { HomePage } from "./homePage/HomePage";
-import { NotFound } from "./nav/NotFound";
-import { MuiThemeProvider } from '@material-ui/core/styles';
-import { createMuiTheme } from '@material-ui/core/styles';
-import { Theme } from "../style/theme";
-import { CssBaseline, withStyles } from "@material-ui/core";
-import { SearchBar as OrgSearchBar } from "./organizations/SearchBar";
-import { SearchBar as VolunteeringSearchBar } from "./volunteering/SearchBar";
+import { withStyles } from "@material-ui/core";
 import { CelebrationModal } from "./modals/CelebrationModal";
-import { FacebookLoginLogout } from "./facebook/FacebookLoginLogout";
 import classNames from 'classnames';
-import { ViewOrganization } from "./organizations/ViewOrganization";
-import { Switch, regexMatch } from "../utils/Switch";
 import { Url } from "../utils/url";
-import { CreateOrganization } from "./organizations/CreateOrganization";
 import { Organization } from "./organizations/organization";
-import { CastleRisk } from "./castleRisk/CastleRisk";
 import { PossibleNavBars } from "./nav/PossibleNavBars";
+import { CurrentRoute } from "./nav/CurrentRoute";
+import { rootStyles } from "../style/rootStyles";
 
 type Props = {
   url:Url
   organizations: Organization[]
   [others:string]:any
+  navTo
 }
 
 export const SiteWithoutStyle = (props:Props) => {
-  const theme = createMuiTheme(Theme);
-  const {classes, shouldShowCelebration, onCelebrationXClicked, isSideNavOpen, url} = props;
-  const { castleRisk } = props
-  const mainPage = Switch
-    .when(url.path)
-    .case("/", <HomePage />)
-    .case("/dev", <FacebookLoginLogout {...props} />)
-    .case("/castleRisk", <CastleRisk {...castleRisk} {...props} />)
-    .case("/organizations/search", <OrgSearchBar {...props} />)
-    .case("/volunteering/search", <VolunteeringSearchBar {...props} />)
-    .case("/organizations/new", <CreateOrganization {...props} />)
-    .matches(regexMatch("/organizations/.+"), <ViewOrganization href={url.path} {...props} />)
-    .result
-    .valueOrDefault(<NotFound />)
-
+  const { shouldShowCelebration, onCelebrationXClicked, isSideNavOpen, url} = props;
+  const { classes, ...propsWithoutClasses } = props;
   return <React.Fragment>
-    <CssBaseline />
-    <div className={classes.root}>
-      <main className={classNames(classes.content, { [classes.contentShift]: isSideNavOpen,})}>
-      <MuiThemeProvider {...{theme}}>
-        <PossibleNavBars {...{history}} {...props} />
-        {mainPage}
-        { shouldShowCelebration && <CelebrationModal numHeartpointsAwarded={10} onXClicked={onCelebrationXClicked} /> }
-      </MuiThemeProvider>
-      </main>
-    </div>
+    <main className={classNames(classes.content, { [classes.contentShift]: isSideNavOpen })}>
+      <PossibleNavBars {...{history}} {...propsWithoutClasses} />
+      {CurrentRoute(url, propsWithoutClasses)}
```

Comment:
main thing here was to remove unneeded theme / css code, which got rid of lots of material ui warnings in the console.

moved the route switching part out into its own thing to keep this file smaller